File: LeetCode/valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        
        //Row Check
        for (int i=0; i<board.length; i++){
            Map<Character, Integer> numberFrequency = new HashMap<>();
            for (int j=0; j<board[i].length; j++){
                if (Character.isDigit(board[i][j])){
                    numberFrequency.put(board[i][j], numberFrequency.getOrDefault(board[i][j],0) + 1);
                }
            }
            for (int k : numberFrequency.values()){
                if (k>1){
                    System.out.print("Stuck in row check");
                    return false;
                }
            }
        }

        //Column Check
        for (int j=0; j<board[0].length; j++){
            Map<Character, Integer> numberFrequency = new HashMap<>();
            for (int i=0; i<board.length; i++){
                if (Character.isDigit(board[i][j])){
                    numberFrequency.put(board[i][j], numberFrequency.getOrDefault(board[i][j],0) + 1);
                }
            }
            for (int k : numberFrequency.values()){
                if (k>1){
                    System.out.print("Stuck in row check");
                    return false;
                }
            }
        }

        // Sub-Box Check
        for (int area = 0; area < 9; area++) {
            Map<Character, Integer> numberFrequency = new HashMap<>();
            
            int rowStart = (area / 3) * 3;
            int colStart = (area % 3) * 3;

            for (int i = 0; i < 3; i++) {
                for (int j = 0; j < 3; j++) {
                    if (Character.isDigit(board[rowStart + i][colStart + j])) {
                        numberFrequency.put(board[rowStart + i][colStart + j], numberFrequency.getOrDefault(board[rowStart + i][colStart + j], 0) + 1);
                    }
                }
            }

            for (int k : numberFrequency.values()) {
                if (k > 1) {
                    System.out.print("Stuck in grid check");
                    return false;
                }
            }
        }
        return true;
    }
}

File: valid_sudoku.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        int rowSize = board.size();
        int colSize = board[0].size();

        // Row Check
        for (int i = 0; i < rowSize; i++) {
            unordered_map<char, int> freqMap;
            for (int j = 0; j < colSize; j++) {
                if (board[i][j] != '.' &&
                    freqMap.find(board[i][j]) != freqMap.end()) {
                    return false;
                }
                freqMap[board[i][j]]++;
            }
        }

        // Col Check
        for (int i = 0; i < colSize; i++) {
            unordered_map<char, int> freqMap;
            for (int j = 0; j < rowSize; j++) {
                if (board[j][i] != '.' &&
                    freqMap.find(board[j][i]) != freqMap.end()) {
                    return false;
                }
                freqMap[board[j][i]]++;
            }
        }

        // Box Check
        for (int boxRow = 0; boxRow < 3; boxRow++) {
            for (int boxCol = 0; boxCol < 3; boxCol++) {
                unordered_map<char, int> freqMap;
                for (int i = boxRow * 3; i < boxRow * 3 + 3; i++) {
                    for (int j = boxCol * 3; j < boxCol * 3 + 3; j++) {
                        if (board[i][j] != '.' &&
                            freqMap.find(board[i][j]) != freqMap.end()) {
                            return false;
                        }
                        freqMap[board[i][j]]++;
                    }
                }
            }
        }

        return true;
    }
};
